### PR TITLE
Sphinx: use html_baseurl for setting the canonical URL

### PR DESCRIPTION
**Pull request by stsewd**
Fixes https://github.com/readthedocs/readthedocs-sphinx-ext/issues/83

### CODE REVIEW

**Comment by stsewd** [comment]
We need to merge and release https://github.com/readthedocs/readthedocs-sphinx-ext/issues/83 first

**Review by ericholscher — APPROVED** [review]
Looks good -- we should probably deprecate the canonical setting at some point, if we have a good way for users to set it.

File: readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl [hunk 1]
```diff
@@ -120,6 +120,14 @@ context = {
     'commit': {% if project.repo_type == 'git' %}'{{ commit|slice:"8" }}'{% else %}'{{ commit }}'{% endif %},
 }
 
+# For sphinx >=1.8 we can use html_baseurl to set the canonical URL.
```
Comment (ericholscher) [inline readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl:123]: We should probably update the docs on this as well.

Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl:123]: Do you mean this https://github.com/readthedocs/readthedocs.org/blob/master/docs/guides/canonical.rst? I didn't find anything else with canonical domain

Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl:123]: I have updated that guide, let me know if that's ok

**Comment by stsewd** [comment]
We need to remove this flag before merging this, so all users have the latest version of our extension https://github.com/readthedocs/readthedocs.org/blob/32510fd0a0930858c0e70e80f0f9a0ff6cdcd24c/readthedocs/doc_builder/python_environments.py#L385-L385

I'll check how many projects we have with that flag

File: docs/guides/canonical.rst [hunk 2]
```diff
@@ -31,6 +31,16 @@ Check your :guilabel:`Admin` > :guilabel:`Domains` page for the domains that we
 Implementation
 --------------
 
+If you are using :doc:`Sphinx </intro/getting-started-with-sphinx>`,
+Read the Docs will set the value of the html_baseurl_ setting (if isn't already set) to your canonical domain.
+
+.. _html_baseurl: https://www.sphinx-doc.org/page/usage/configuration.html#confval-html_baseurl
+
+If you are using :doc:`MkDocs </intro/getting-started-with-mkdocs>`,
+you can use the site_url_ setting.
```
Comment (ericholscher) [inline docs/guides/canonical.rst:40]: Do we not automatically support this at all for mkdocs?

Comment (stsewd) [reply docs/guides/canonical.rst:40]: No, but we can do it if we want https://github.com/readthedocs/readthedocs.org/issues/4820 :)

**Comment by stsewd** [comment]
This is also blocked on https://github.com/readthedocs/readthedocs.org/blob/2a519f1146142d18f6a63b61c2f08984067280e0/readthedocs/doc_builder/python_environments.py#L383-L383

https://github.com/readthedocs/sphinx_rtd_theme/pull/1003

**Comment by mgeier** [comment]
Is there any news on this?
Can this be merged?

**Comment by stsewd** [comment]
We are already using the latest version of the rtd-sphinx-ext, but we still need to merge and release https://github.com/readthedocs/sphinx_rtd_theme/pull/1003/ first.

**Comment by mgeier** [comment]
Thanks for the update!

**Comment by stsewd** [comment]
A new release of the t heme is out, we can merge this now

**Comment by ericholscher** [comment]
I've confirmed this is working now on a doc rebuild with our latest deploy.

**Pull request merged**